### DISCUSSION TITLE
RFC - Removal of the Sidebar and Topbar components from admin-ui

### DISCUSSION THREAD

**Discussion by marcelovicentegc** [vtex/shoreline#571] [Technical RFCs]
# RFC - Removal of the Sidebar and Topbar components from admin-ui

## Problem
Currently, we have two components that are somehow hidden on the library and are only used on the admin-shell: the Sidebar and the Topbar components.

Whenever changes are required for these components, these demands come from adaptations to the admin-shell. They were thought and implemented for this specific application, and not for general use such as the Button, DataGrid and Page components for instance.

There is little to no use for them to live within this package, and the price we pay for having them live within admin-ui is high, since we can't directly make changes to them on the only application that uses them: the admin-shell. 

If a change on the Sidebar or Topbar of the admin-shell is required, this is what we do:
- Implement these changes on the admin-ui
- Release a new version of the admin-ui
- Update admin-ui to the latest version on the admin-shell so we can bring the changes
- Release a new version of the admin-shell

When it could simply be:
- Implement the changes directly on the admin-shell
- Release a new version of the admin-shell

## Proposal
Remove the Sidebar and Topbar components from admin-ui, and bring them to the admin-shell. In case we decide that we want to expose these components to the world, [we should then properly document and reference it on admin-ui documentation](https://admin-ui.vercel.app/).

### Pros
- admin-ui gets more lightweight 
- Development on the admin-shell is accelerated whenever changes to the Sidebar and Topbar are required

### Cons
- The effort of removing the components from the library
- We will not be able to reuse the Sidebar and Topbar somewhere else

### Unresolved questions
Does it make sense to incorporate the Sidebar and Topbar components into the admin-ui library?

**Comment by lucasaarcoverde** [comment]
> Does it make sense to incorporate the Sidebar and Topbar components into the admin-ui library?

In the past we had some motivation to move these components for the admin-ui, for example:
* Developing a remake of the sidebar in the admin-shell while using `vtex link` was very slow
* The admin-ui was very immature, so sometimes we had to wait for updates on the lib to be able to implement a desired behavior in the admin-shell

Now I see that we have a more mature lib and I agree to move both components to the admin-shell. IMO they are very specific, and we only had the intention of developing it to be used by the admin until now.

One thing about being used only on the admin-shell, [it looks like we have another project using it as well](https://github.com/vtex/povo-page/blob/main/src/components/Sidebar/index.tsx#L1-L11). We should double-check and warn them if necessary and we decide to remove it

**Reply by marcelovicentegc** [reply]
> Developing a remake of the sidebar in the admin-shell while using vtex link was very slow

Excellent point! Now that we are moving outside of Render the development experience actually becomes better (for these components) than admin-ui's

> One thing about being used only on the admin-shell, [it looks like we have another project using it as well](https://github.com/vtex/povo-page/blob/main/src/components/Sidebar/index.tsx#L1-L11). We should double-check and warn them if necessary and we decide to remove it

Good catch @lucasaarcoverde. I will have them on my radar if we move forward with the migration.

**Comment by beatrizmilhomem** [comment]
The migration does make sense to me. The sidebar and topbar exist as any other Admin module that is constructed based on the Admin UI and can evolve/change according to user or business needs.

**Comment by matheusps** [comment]
Another con is that we will not be able to reuse it somewhere else. Keep in mind that generally design systems keep these global components on their codebase since they're an essential part of the product. For reference:

- IBM Carbon: https://carbondesignsystem.com/patterns/global-header/#related. 
- Atlassian DS: https://atlassian.design/components/atlassian-navigation/examples. 
- Shopify Polaris: https://polaris.shopify.com/components/top-bar.

**Reply by marcelovicentegc** [reply]
Good catch! Just added that `We will not be able to reuse the Sidebar and Topbar somewhere else` as another con! 

I've got a counter point regarding that:
- Do we foresee other teams or projects using these components?

**Comment by marcelovicentegc** [comment]
This is being addressed by:
- https://github.com/vtex/admin-shell-widgets/pull/13
- https://github.com/vtex/admin-ui/pull/610

**Comment by davicostalf** [comment]
@matheusps @marcelovicentegc Can we close this?

**Comment by marcelovicentegc** [comment]
Solved by https://github.com/vtex/admin-ui/pull/610
